src/giskardpy/world.py

```diff
@@ -161,7 +161,6 @@ def attach_existing_obj_to_robot(self, name, link, pose):
         """
         # TODO this should know the object pose and not require it as input
         self._robot.attach_urdf_object(self.get_object(name), link, pose)
-        self.remove_object(name)
```

Comment:
Oh I don't like that. Attaching should definitely remove the object. The world class should work properly on its own.